### ISSUE TITLE
Question/comment for jpeg corrupted data issues

### ISSUE THREAD

**Issue by naatje80** [TurboVNC/turbovnc#222]
Some of our users were encountering jpeg corrupted data errors in combination with the usage of the IC Design software Cadence Virtuoso (depending on their usage this could occur once or a couple of times a day). 

I found an older thread in which you clearly describe this cause of issue:
https://sourceforge.net/p/turbovnc/mailman/message/34937327/
However, I was wondering if this is still valid at the moment or not. I've checked, and TigerVNC nowadays seems to be compiled with libjpeg-turbo (the documentation strongly advises to compile with libjpeg-turbo instead of the older libjpeg library). 

In our case we seem to have fixed the issue by compiling the TigerVNC server with the latest turbo-jpeg library, instead of the older version supplied on EL7. So hopefully this is also helpful for people that are encountering similar issues.

**Comment by dcommander** [comment]
Yeah, I have no idea why Red Hat continues to ship a beta version of libjpeg-turbo with an operating system that is supposed to be enterprise stable.  They have back-ported some of the security fixes introduced since then but not all, and there are certainly a lot of other bug fixes they have not back-ported.  If they want to standardize around libjpeg-turbo 1.3.x, that's fine.  I already maintain a branch for that, and I'm happy to cherry pick any relevant bug fixes into that branch.  But beta releases cease to be supported or supportable after the corresponding major version is released.  They should at the very least be using 1.3.1 and back-porting relevant fixes onto that.

**Comment by dcommander** [comment]
If the issue went away by upgrading libjpeg-turbo, then it wasn't likely the same issue reported in that old thread.  That issue was due to the TigerVNC developers insisting that I modify the 25-year-old libjpeg API to suit their less-than-robust usage of it rather than making their usage of it more robust.  I refused, and I have no idea whether that issue was ever resolved within TigerVNC.  The quality (or lack thereof) of other people's VNC servers isn't my problem.
